### PR TITLE
Add first class support for LiteLLM proxy and LM Studio

### PR DESCRIPTION
**Pull request by abhishekgml1991**
## Description

[Provide a brief description of the changes in this PR]

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Is there an addition/change in API Request, Response? If yes, documentation is required)
- [ ] Performance improvement
- [ ] Code refactoring
- [ ] Security fix

## Related Issues

- Fixes #[issue number]
- Closes #[issue number]
- Related to #[issue number]

## How Has This Been Tested?

[Describe the tests you ran to verify your changes. Provide instructions so we can reproduce.]

### Test Configuration

- [x] Unit tests pass
- [x] Integration tests pass
- [ ] Manual testing completed
- [ ] Cross-browser testing (if applicable)
- [ ] Mobile responsiveness tested (if applicable)

## Core Functionality Testing

Please confirm that the following core functionalities are working as expected:

- [x] **Search capabilities** are working correctly
- [x] **Knowledge search** is functioning properly
- [ ] **Connector indexing** is working as expected
- [x] **Citations** are displaying and linking correctly
- [ ] **Documentation** is updated at https://docs.pipeshub.com/introduction

## What You Have Tested

Please describe what you have specifically tested:

- [ ] Feature works in development environment
- [ ] Feature works in staging environment
- [ ] Error handling scenarios tested
- [ ] Edge cases considered and tested
- [ ] Performance impact assessed
- [ ] Security implications reviewed

### Test Results

[Provide details of your test results here]

## Screenshots/Videos

**Before:**
[Attach screenshots/videos showing the current behavior]

**After:**
[Attach screenshots/videos showing the new behavior]

**Note:** Screenshots are required before merge for UI changes.

## Code Quality Checklist

- [ ] My code follows the project's coding standards
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Documentation

- [ ] Documentation has been updated (if applicable)
- [ ] API documentation updated (if applicable)
- [ ] README updated (if applicable)
- [ ] Changelog updated (if applicable)

## Security Considerations

- [ ] No sensitive data is exposed
- [ ] Input validation is implemented where necessary
- [ ] Authentication/authorization is properly handled
- [ ] No security vulnerabilities introduced

## Breaking Changes

- [ ] This PR introduces breaking changes
- [ ] Migration guide provided (if applicable)
- [ ] Version bump required

If breaking changes are introduced, please describe them here:

[Describe any breaking changes and how users should adapt]

## Performance Impact

- [ ] No performance impact
- [ ] Performance improved
- [ ] Performance impact assessed and acceptable

[Describe any performance implications]

## Dependencies

- [ ] No new dependencies added
- [ ] New dependencies are necessary and approved
- [ ] Dependencies updated and tested

List any new dependencies:
- [Dependency name and version]

## Deployment Notes

[Any special deployment considerations or steps]

## Checklist Before Merge

- [ ] All tests are passing
- [ ] Code review completed and approved
- [ ] Documentation updated and reviewed
- [ ] Screenshots/videos attached for UI changes
- [ ] Core functionality verified
- [ ] Security review completed (if applicable)
- [ ] Performance impact assessed
- [ ] Ready for production deployment

## Additional Notes

[Any additional information that would be helpful for reviewers]

---

**For Reviewers:**

Please ensure all checklist items are completed before approving this PR. Pay special attention to:
- Core functionality testing
- Security implications
- Performance impact
- Documentation completeness

### CODE REVIEW

File: backend/python/app/api/routes/health.py [hunk 1]
```diff
@@ -920,6 +920,17 @@ async def perform_image_generation_health_check(
...
+            endpoint = configuration.get("endpoint", "").rstrip("/")
+            async with httpx.AsyncClient(timeout=30.0) as http_client:
+                resp = await http_client.get(
+                    f"{endpoint}/health",
+                    headers={"Authorization": f"Bearer {configuration['apiKey']}"},
+                )
+                if resp.status_code >= 500:
+                    raise RuntimeError(
+                        f"LiteLLM Proxy health check returned HTTP {resp.status_code}"
+                    )
```
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/python/app/api/routes/health.py:934]: Fixed

File: backend/python/app/api/routes/health.py [hunk 2]
```diff
@@ -1031,6 +1042,17 @@ async def perform_tts_health_check(
...
```
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/python/app/api/routes/health.py:1057]: Fixed

File: backend/python/app/api/routes/health.py [hunk 3]
```diff
@@ -1191,6 +1213,17 @@ async def perform_stt_health_check(
...
```
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/python/app/api/routes/health.py:1229]: Fixed

File: backend/python/app/utils/aimodels.py [hunk 4]
```diff
@@ -347,6 +354,30 @@ def get_embedding_model(provider: str, config: dict[str, Any], model_name: str |
...
+
+    elif provider == EmbeddingProvider.LITELLM_PROXY.value:
+        from langchain_openai.embeddings import OpenAIEmbeddings
+
+        llp_emb_kwargs: Dict[str, Any] = dict(
+            model=model_name,
+            api_key=configuration['apiKey'],
+            base_url=configuration["endpoint"],
+            check_embedding_ctx_length=False,
+        )
```
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/python/app/utils/aimodels.py:377]: Fixed

File: backend/python/app/utils/aimodels.py [hunk 5]
```diff
@@ -1065,6 +1128,14 @@ def get_image_generation_model(
             api_key=configuration["apiKey"],
         )
 
+    if provider == ImageGenerationProvider.LITELLM_PROXY.value:
+        return _OpenAIImageAdapter(
+            model=model_name,
+            api_key=configuration["apiKey"],
+            base_url=configuration["endpoint"],
+            provider_override=ImageGenerationProvider.LITELLM_PROXY.value,
+        )
```
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/python/app/utils/aimodels.py:1137]: Fixed

File: backend/python/app/utils/aimodels.py [hunk 6]
```diff
@@ -697,6 +728,32 @@ def get_generator_model(provider: str, config: dict[str, Any], model_name: str |
...
+        from langchain_openai import ChatOpenAI
+        is_reasoning_model = "gpt-5" in model_name or config.get("isReasoning", False)
+        temperature = 1 if is_reasoning_model else configuration.get("temperature", 0.2)
+        return ChatOpenAI(
+                model=model_name,
+                temperature=temperature,
+                timeout=DEFAULT_LLM_TIMEOUT,
+                api_key=configuration["apiKey"],
+                base_url=configuration["endpoint"],
+                stream_usage=True,
```
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/python/app/utils/aimodels.py:754]: Fixed

File: backend/python/app/utils/aimodels.py [hunk 7]
```diff
@@ -1427,6 +1498,16 @@ def get_tts_model(
...
 
+    if provider == TTSProvider.LITELLM_PROXY.value:
+        return _OpenAITTSAdapter(
+            model=model_name,
+            api_key=configuration["apiKey"],
+            default_voice=configuration.get("voice"),
+            default_format=configuration.get("responseFormat"),
+            base_url=configuration["endpoint"],
+            provider_override=TTSProvider.LITELLM_PROXY.value,
+        )
```
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/python/app/utils/aimodels.py:1509]: Fixed

File: backend/python/app/utils/aimodels.py [hunk 8]
```diff
@@ -2078,4 +2165,12 @@ def get_stt_model(
             api_key=configuration["apiKey"],
         )
 
+    if provider == STTProvider.LITELLM_PROXY.value:
+        return _OpenAISTTAdapter(
+            model=model_name,
+            api_key=configuration["apiKey"],
+            base_url=configuration["endpoint"],
+            provider_override=STTProvider.LITELLM_PROXY.value,
+        )
```
[the remark this answers is not in the document]
Comment (abhishek-pipeshub) [reply backend/python/app/utils/aimodels.py:2174]: Fixed

**Review by abhishek-pipeshub — APPROVED** [review]
